Polygon_mesh_processing/examples/Polygon_mesh_processing/hole_filling_example_SM.cpp

```diff
@@ -44,6 +47,76 @@ bool is_small_hole(halfedge_descriptor h, Mesh & mesh,
   return true;
 }
 
+struct Stop : std::exception
+{
+    Stop()
+    {}
+};
+
+struct Progress {
+
+    Progress(double time_limit)
+        : time_limit(time_limit)
+    {}
+
+    Progress(const Progress&) = delete;
+
+      void start_quadratic_phase(int n)
+      {
+        timer.start();
+        quadratic_n = n;
+        quadratic_report = n / 10;
+        std::cout << "Start Quadratic phase with estimated " << n << " steps" << std::endl;
+      }
+
+      void quadratic_step()
```

Comment:
If I understand correctly, then quadratic_step() can be called more than once for finite facet. But in this case, we will get the wrong percentage of progress.

I slightly modified one of the methods:

```
void end_quadratic_phase(bool success) const
{
    std::cout << "End Quadratic phase " << (success ? "(success)" : "(falied)") << std::endl;
    std::cout << "quadratic_i: " << quadratic_i << std::endl;
}
```

And this is the output I get:

```
Start Quadratic phase with 159 steps
10.0629%
19.4969%
28.9308%
38.3648% 
47.7987%
57.2327%
66.6667%
76.1006%
85.5346%
94.9686%
104.403%
113.836%
123.27%
132.704%
142.138%
151.572%
161.006%
170.44%
179.874%
189.308%
198.742%
End Quadratic phase (success)
quadratic_i: 319
* Number of facets in constructed patch: 68
  Number of vertices in constructed patch: 24
  Is fairing successful: 1
```

Reply:
`quadratic_i`  is the number of times the function `quadratic_step()`  is called.    `quadratic_n`   is the maximal number that `quadratic_step()` gets called.  In practice it will always be lower, so we are conservative.

Reply:
`quadratic_i`  is the number of times the function `quadratic_step()`  is called.    `quadratic_n`   is the maximal number that `quadratic_step()` gets called.  In practice it will always be lower, so we are conservative.

Reply:
You are right.  I just fixed it.    We count the number of recursive function calls.    I moved the counting callback after the test which tests if there is something to do.  The change in the example itself is unrelated to your observation but just a cleanup.

Reply:
Thanks! Now progress works like this:
```
Start Quadratic phase with 6107 steps
10.0049%
19.9935%
29.982%
39.9705%
End Quadratic phase (success)
quadratic_i: 2660
```
Of course, this is probably also not what the user expects, but probably in the case of recursive calls, it is impossible to know in advance how many of them there will be.